### PR TITLE
#169535991 implements a create entry endpoint

### PR DESCRIPTION
**Pull request by bbaime98**
#### What does this PR do?
- implements a create entry endpoint
#### Description of Task to be completed?
- add a create entry route
- add entry inputs validation
- add a create entry function
- add tests for create entry endpoint
#### How should this be manually tested?
- cretate an account first
- then create an entry 
#### What are the relevant pivotal tracker stories?
-  [169535991](https://www.pivotaltracker.com/story/show/169535991)
#### Screenshots (if appropriate)
image

#### Questions

### CODE REVIEW

File: server/tests/db-entry-mockdata.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+const entries = {
```
Comment (ekumamait) [inline server/tests/db-entry-mockdata.js:1]: Do we still require this, if yes for what?

File: server/tests/db-entryTests.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        res.body.data.should.have.property('description').eql('new entry new entry new entry new entry');
+        res.body.data.should.have.property('entryid');
+        res.body.data.should.have.property('createdon');
+        done();
+      });
+  });
+  it('should return no Invalid token ', (done) => {
+    chai.request(app)
+      .post('/api/v2/entries/')
+      .set('token', 'eyJhbGcIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MSwiZW1haWwiOiJyZWxseUBnbWFpbC5jb20iLCJpYXQiOjE1NzI0Njg2NTJ9.eXMdkYoWmxaCxd8mXOYOc6crWKdBXcEcLpuSBwySBvI')
```
Comment (ekumamait) [inline server/tests/db-entryTests.js]: How about we move this data as well

File: server/tests/db-entry-mockdata.js (hunk 1 above)
Comment (bbaime98) [reply server/tests/db-entry-mockdata.js:1]: yes, am using them for testing, As I deleted the users table,  a new user has should create an account first to create an entry

File: server/tests/db-entryTests.js (hunk 2 above)
Comment (bbaime98) [reply server/tests/db-entryTests.js]: you mean, I should move token to mockdata?

Comment (ekumamait) [reply server/tests/db-entryTests.js]: Yes can't we do that?

Comment (bbaime98) [reply server/tests/db-entryTests.js]: yes we can. I did it. check on my new commit

**Pull request merged**